core/src/dynamic_widget.rs

```diff
@@ -80,11 +93,70 @@ impl<D: 'static> Render for DynRender<D> {
     self.self_render.hit_test(ctx, pos)
   }
 
-  fn can_overflow(&self) -> bool { self.self_render.can_overflow() }
+  fn can_overflow(&self) -> bool {
+    !self.drop_until_widgets.is_empty() || self.self_render.can_overflow()
+  }
 
   fn get_transform(&self) -> Option<Transform> { self.self_render.get_transform() }
 }
 
+#[derive(Default)]
+struct WidgetsHost {
+  wids: RefCell<HashSet<WidgetId>>,
+}
+
+impl WidgetsHost {
+  fn add(&self, wid: WidgetId) { self.wids.borrow_mut().insert(wid); }
+
+  fn is_empty(&self) -> bool { self.wids.borrow().is_empty() }
+
+  fn paint(&self, ctx: &mut PaintingCtx) {
+    self.wids.borrow().iter().for_each(|wid| {
+      wid.paint_subtree(ctx.arena, ctx.store, ctx.wnd_ctx, ctx.painter);
+    });
+  }
+
+  fn remove_unhost(&self, arena: &mut TreeArena, store: &mut LayoutStore) {
+    let mut set = HashSet::new();
+    let arr = self
+      .wids
+      .borrow()
+      .iter()
+      .filter(|id| {
+        query_drop_until_widget(**id, arena).map_or(true, |w| w.state_ref().delay_drop_until)
+      })
+      .cloned()
+      .collect::<Vec<_>>();
+    arr.iter().for_each(|id| {
+      self.wids.borrow_mut().remove(id);
+    });
+    self.collect_subtree(arr.into_iter(), arena, &mut set);
+    set.iter().for_each(|id| id.remove_subtree(arena, store));
+  }
+
+  fn collect_subtree(
+    &self,
+    it: impl Iterator<Item = WidgetId>,
+    arena: &TreeArena,
+    set: &mut HashSet<WidgetId>,
+  ) {
+    it.for_each(|id| {
+      id.descendants(arena).for_each(|wid| {
+        wid
+          .assert_get(arena)
+          .query_on_first_type(QueryOrder::OutsideFirst, |w: &WidgetsHost| {
+            w.collect_subtree(w.wids.borrow().iter().cloned(), arena, set)
```

Comment:
Seems unnecessary recursive collect.

Reply:
necessary when a delay drop widget is hosted by another delay drop widget